core/src/subject/SubjectEntity.ts

```diff
@@ -221,10 +232,62 @@ export class SubjectEntity {
 
     return []
   }
+
+  static async query(perspective: PerspectiveProxy, query?: SubjectEntityQueryParam) {
```

Comment:
I think it would make sense to actually create a separate SubjectRepo class, or something similar that provides more of the query, findAll, find etc methods. It doesn't quite feel right to have a query method on a SubjectEntity, as it refers more to *one* Entity. What do you think @lucksus ?

Reply:
Still, it is very nice to not worry about more different types of classes. So I don't have a very strong opinion, and I would let this be up to you backend folks!

Reply:
This is a `static` function, so it's a class method, not an object method. See the test below. So I think that is fine.

I think the `source` parameter makes sense here in the query.

Only thing I don't feel good about is enforcing the creation of these source links. And if we do, I think the predicate should at least be `ad4m://something`.

Talked about this with Harlan last week and mentioned a predicate for arbitrary "associations" between objects...